h/static/scripts/directive/annotation.js

```diff
@@ -111,13 +109,20 @@ function updateDomainModel(domainModel, vm, permissions) {
 }
 
 /** Update the view model from the domain model changes. */
-function updateViewModel($scope, time, domainModel, vm, permissions) {
+function updateViewModel($scope, time, domainModel,
+                         vm, permissions) {
 
   vm.form = {
     text: domainModel.text,
     tags: viewModelTagsFromDomainModelTags(domainModel.tags),
   };
-  vm.annotationURI = new URL('/a/' + domainModel.id, vm.serviceUrl).href;
+
+  if (domainModel.links) {
+    vm.annotationURI = domainModel.links.incontext || domainModel.links.html;
```

Comment:
If the default is to be empty-string, then this should probably have a `|| ''` at the end, too.